homeassistant/components/media_player/webostv.py

```diff
@@ -363,3 +381,84 @@ def media_next_track(self):
     def media_previous_track(self):
         """Send the previous track command."""
         self._client.rewind()
+
+    def play_media(self, media_type, media_id, **kwargs):
+        """Play a piece of media."""
+        self._playing = True
+        self._state = STATE_PLAYING
+        if media_type != MEDIA_TYPE_CHANNEL:
+            self._client.play()
+        else:
+            self._client.set_channel(media_id)
+
+    def update_channel_info(self):
+        channel = self._client.get_current_channel()
+
+        # no need to updated channel info while we are not watching TV
+        if channel.get('returnValue') is None:
+            self.reset_channel_info()
+            return
+
+        self._channel = channel.get('channelName')
+        self._title = self._channel
+        self._duration = None
+        channel_id = channel.get('channelId')
+        self._progress_updated_at = util.dt.utcnow()
+        info = self._client.get_channel_info()
+
+        # tv returns the list of all channels
+        now = datetime.now(pytz.UTC)
+        for program in info.get('programList', []):
+            # skip any invalid data
+            if program['startTime'] == '' or program['endTime'] == '':
+                continue
+
+            st = self._date_helper(program['startTime'])
+            ed = self._date_helper(program['endTime'])
+            if program['channelId'] == channel_id and st <= now < ed:
+                self._title = "{}: {}".format(
+                    self._channel,
+                    program['programName']
+                )
+                self._duration = program['duration']
+                self._progress = (now - st).total_seconds()
```

Comment:
Don't update progress unless channel changed. Also, only when you update this property can you update the progress_updated_at property